components/script/script_thread.rs

```diff
@@ -1169,7 +1238,24 @@ impl ScriptThread {
                 ScriptThreadEventCategory::ExitFullscreen => ProfilerCategory::ScriptExitFullscreen,
                 ScriptThreadEventCategory::PerformanceTimelineTask => ProfilerCategory::ScriptPerformanceEvent,
             };
-            profile(profiler_cat, None, self.time_profiler_chan.clone(), f)
+
+            let start = precise_time_ns();
+            let t = profile(profiler_cat, None, self.time_profiler_chan.clone(), f);
+            let end = precise_time_ns();
+            debug!("Task {:?} took {}", category, end - start);       // TODO do we want to do anything with this?
+
+            for (doc_id, doc) in self.documents.borrow().iter() {
+                match p_id {
+                    Some(p_id) => {
+                        if p_id == doc_id && end - start > InteractiveWindow::max_task_time() {
+                            doc.start_tti()
+                        }
+                    },
+                    _ => ()
+                }
+                doc.check_tti();
```

Comment:
Right now this code only executes if we run with `-Z profile-script-events`. We want the TTI check to always execute.

Reply:
I agree-- should i just duplicate the measurement in the else?

Reply:
I would do:
```rust
let start = ...;
let value = if opts::get().profile_script_events {
    ...
    profile(...)
} else {
    f()
};
let end = ...;
for (doc_id, doc) in self.documents.borrow().iter() {
```